libclc/utils/libclc-remangler/LibclcRemangler.cpp

```diff
@@ -465,6 +479,17 @@ class Remangler {
 class TargetTypeReplacements {
   SmallDenseMap<const char *, const char *> ParameterTypeReplacements;
   SmallDenseMap<const char *, const char *> AliasTypeReplacements;
+  SmallDenseMap<const char *, const char *> RemangledAliasTypeReplacements;
+
+  void CreateRemangledTypeReplacements() {
+    for (auto &TypeReplacementPair : ParameterTypeReplacements)
+      if (AliasTypeReplacements.find(TypeReplacementPair.getFirst()) ==
+              AliasTypeReplacements.end() ||
+          AliasTypeReplacements[TypeReplacementPair.getFirst()] !=
+              TypeReplacementPair.getSecond())

```

Comment:
It seems a bit redundant to do the lookup twice.

Reply:
Sorry, that was a mistake on my part. I tried to simplify the conditional statement using if statement order of evaluation but made a mistake.
I have corrected it, to turn it into an if, else if statement.